torchrl/modules/tensordict_module/common.py

```diff
@@ -88,7 +80,7 @@ def _forward_hook_safe_action(module, tensordict_in, tensordict_out):
                 )
 
 
-class TensorDictModule(nn.Module):
+class TensorDictModule(_TensorDictModule):
```

Comment:
I was wondering whether we should consider changing the name here, as it could be confusing if both tensordict and TorchRL have `TensorDictModule` classes. 

I don't have a good name, but I'm thinking something like `TensorDictModuleWithSpec` to convey the difference?

Reply:
Yes that's something I thought about too
Maybe we can drop the TensorDict part (it's implicitly the case since torchrl is all-in into adopting tensordict)
`ModuleWithSpec`? `SpecModule`? `SafeModule`? `RLModule`?

Reply:
Dropping TensorDict part sounds good to me.

Of your suggestions I'm most drawn to `SafeModule` and `SpecModule` in that order I think?

Reply:
Sorry for the slow answer
We can also think about user experience, and for that RLModule seems good
In ReAgent they have a (improperly named) ModelBase class that all models inherit from. It's not only bad because Model-Based in a subfield in RL, but also because it fails to account for what it is (the base class of all policies and value networks).
SafeModule is also good as the main difference is mainly that putting a spec allows for safe sampling / mapping (it's more about safety than about the spec IMO). In that sense RLModule is less self-explanatory.
TLDR let's go for SafeModule :p